src/main/java/org/dependencytrack/policy/cel/CelPolicyQueryManager.java

```diff
@@ -140,35 +140,29 @@ List<ComponentProjection> fetchAllComponents(final long projectId, final Collect
         String sqlSelectColumns = Stream.concat(
                         Stream.of(ComponentProjection.ID_FIELD_MAPPING),
                         getFieldMappings(ComponentProjection.class).stream()
+                                //filtering published_at and latest_version from component requirements as these fields need to
+                                //come from joining with the integrityMetaComponent and RepostioryMetaComponent tables respectively
                                 .filter(mapping -> protoFieldNames.contains(mapping.protoFieldName()))
+                                .filter(fieldMapping -> !fieldMapping.sqlColumnName().equals("PUBLISHED_AT"))
+                                .filter(fieldMapping -> !fieldMapping.sqlColumnName().equals("LATEST_VERSION"))
```

Comment:
These `filter`s can be avoided when removing the `@MappedField` annotations from the respective fields in the projection.